src/FakeItEasy/Expressions/ArgumentConstraints/EqualityArgumentConstraint.cs

```diff
@@ -32,7 +36,18 @@ public override string ToString()
                 return "\"{0}\"".FormatInvariant(stringValue);
             }
 
-            return this.ExpectedValue.ToString();
+            try
+            {
+                return this.ExpectedValue.ToString();
+            }
+            catch (Exception)
+            {
+                var type = this.ExpectedValue.GetType();
+
+                return type.Namespace == "Castle.Proxies"
+                    ? $"Faked {type.GetTypeInfo().BaseType.FullName}"

```

Comment:
For a faked interface, this will print "Faked System.Object", rather than the faked interface.
And it misses faked delegates altogether.

Consider trying to get the FakeManager (Unfortunately we currently only have a `GetFakeManager` that throws when we call that on a non-Fake, as far as I can tell. But I see no reason not to expose an internal accessor that isn't so cranky.)
If `ExpectedValue` has a FakeManager, we can get the faked type from `FakeObjectType` and slap a "Fake " on the front of it.

While we're at it, the specs should probably include tests for faked interfaces, classes, and delegates, just to be sure we catch all the wacky cases.